analyzers/src/SonarAnalyzer.Common/AnalysisContext/SonarAnalysisContextBase.cs

```diff
@@ -125,7 +125,10 @@ public bool HasMatchingScope(DiagnosticDescriptor descriptor)
             descriptor.CustomTags.Contains(tag);
     }
 
-    private bool ShouldAnalyzeFile(SonarLintXmlReader sonarLintXml, string filePath) =>
+    private bool ShouldAnalyzeFile(SonarLintXmlReader sonarLintXml, SyntaxTree tree) =>
+        !IsUnchanged(tree) && IsInSonarLintContextAndIncluded(sonarLintXml, tree.FilePath);
```

Comment:
The `IsInSonarLintContextAndIncluded` is not right. The full method reads `we should analyze file if it isn't known to be unchanged AND is only in SonarLint AND included`.
That is not true, as non-sonarlint should also be analyzed.